### DISCUSSION TITLE
Different forecasts for same time series w/ NBEATS

### DISCUSSION THREAD

**Discussion by MeirYosub** [Nixtla/neuralforecast#969] [Q&A] [answered]
Hi there,
In addition to the statistical forecasting using the ARIMA and ETS models I try to implement forecasting using neural networks.
I encountered an unexplained matter that when I make a forecast with the NBEATS model for a single time series with a certain unique_id I get completely different forecasting results than those obtained when the forecast is made for the same series and others with different unique_id at the same time.
What is the reason for this?

I must point out that Nixtla has done a very impressive job in building and implementing the models for predicting time series and there is no doubt that there is a significant contribution in capabilities to the entire community. 
Huge thumbs up for that.

**Comment by elephaint** [comment]
Hi, thanks for using neuralforecast and thanks for the kind words :) 

Unfortunately without a standalone piece of code that I can run myself that exposes the issue it's difficult to help you; it may be an error in Neuralforecast or NBEATS, or an error in your code / training pipeline, it's near-impossible to tell without more information. So, if you can provide a minimal example that reproduces this behaviour with NBEATS I can help you further.

**Reply by MeirYosub** [reply]
code demo and the results

data

![image](https://github.com/Nixtla/neuralforecast/assets/100159578/7113acc2-41f5-4c74-bd86-276d441261dc)


model fit and forecast

![image](https://github.com/Nixtla/neuralforecast/assets/100159578/99889b09-4b6e-4537-89cf-d1ecf1460ba8)


subset the data to only 2 unique_id series

![image](https://github.com/Nixtla/neuralforecast/assets/100159578/5cffbd70-3ad5-469a-81d5-ec99bb09b370)

getting different forecast

![image](https://github.com/Nixtla/neuralforecast/assets/100159578/91fd45a5-00f0-4d42-8f89-3401debff553)

**Reply by MeirYosub** [reply]
Hi there,
Anything new on this topic?

**Reply by elephaint** [reply]
Hi - it's hard to judge from pictures (a standalone piece of code that I can run myself would help).

That said, first you are training a model on many series, thereafter only on 2. So obviously you will get different results (?)

**Reply by MeirYosub** [reply]
Hi there
 I am adding code (jupyter notebook as txt file)  + data

[NBEATS_demo (copy).txt](https://github.com/Nixtla/neuralforecast/files/15261636/NBEATS_demo.copy.txt)
[data.csv](https://github.com/Nixtla/neuralforecast/files/15261639/data.csv)

**Comment by MeirYosub** [comment]
Something here is not clear to me.
Why forecasting several series simultaneously with the NBEATS model should change the forecast result of a specific series? 
After all, each series is actually separate from the others and is identified by a unique key.
This behavior does not exist when forecasting with statistical models such as ETS and autoARIMA.

**Reply by MeirYosub** [reply]
Any new comment on the issue ?

**Reply by MeirYosub** [reply, marked as answer]
From what I am able to understand from the results: in the case of several time series, one and only neural model (NBEATS) is adjusted **for all the series**, in contrast to the statistical models in which a separate model is adjusted for each series.

**Comment by elephaint** [comment]
@MeirYosub Your understanding is (mostly) correct.

If you are training a single neural network on multiple series, you get a single model that can produce forecasts for multiple series. In statistical methods we typically train a single model per series. A key motivation of using a neural network is to leverage its superior representation capacity to learn a single model across many series such that it can be used for forecasting multiple series. Training a neural network on a single series only (and thus having separate models per series) kind of destroys the entire reason of using a neural network.

I'm still not sure there is an issue here (it seems mostly a question of understanding the basic concepts of forecasting with neural networks). Do you have the feeling you understand these concepts? If not, happy to provide further clarifications. If so, I think we can close this issue. Let me know :)

**Reply by MeirYosub** [reply]
Many thanks for the explanation. The issue seems to have been understood.
